File: Leetcode/Remove Nth Node From End of List/Remove Nth Node From End of List.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {

        // Creating a new variable temp of type ListNode
        ListNode temp = new ListNode();
        temp = head;
        int len = 0;
        // Pointing temp to the end of the linked list
        while(temp != null){
            len++;
            temp = temp.next;
        }
        // Effective value of n from the start of the list
        int eff = len - n + 1;
        ListNode p = head;
        int count = 1;
        ListNode q = head;
        if(len == 1)
            return temp;
        
        if(eff == 1){
            return head.next;
        }
        
        // Traverse till we reach one node before the Nth node.
        while(count < eff - 1){
            p = p.next;
            count++;
        }
        // Bypassing the node
        p.next = p.next.next;
        return head;
    }
}

File: Leetcode/Remove Nth Node From End of List/Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int count=0;
        ListNode*p=head;

        //advance the pointer to the end to count the number of nodes
        while(p){
            count++;
            p=p->next;
        }

        // if count is 1, then only one node is present, just delete that
        if (count==1){
            delete p;
            return NULL;
        }


        p=head;
        int i=count-n-1;

        if (n==count){
            head=head->next;
            delete p;
            return head;
        }

        //advance the pointer to one node before the node to be deleted.
        while(i--){
            p=p->next;
        }
        //deletion of node.
        ListNode*t=p->next;
        p->next=t->next;
        delete t;
        return head;
    }
};